### PR TITLE
test(hx-popover): component tests surf2016

### PR DESCRIPTION
**Pull request by lalithkarikelli**
## Description

Test Case `<hx-popover>`

image

### What are the relevant story cards/tickets? Any additional PRs or other references?

Jira: SURF 2016

## Before you request a review for this PR:

- [x] Did you add component tests for any new code?
- [x] Did you run the component unit tests via `yarn test` to ensure all tests passed?
- [x] Did you include a screenshot of the component tests?
- [ ] If you changed/added functionality, did you update the demo page and documentation?
- [ ] If needed, did you add or modify the demo test page to test the changed/added functionality?
- [x] Did you assign reviewers?
- [x] In Jira, have you linked to this PR on the ticket(s)?

### CODE REVIEW

File: src/elements/hx-popover/index.spec.js [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+                    expect(name).to.be.equal('');
+                } else {
+                    expect(name).to.be.null;  // IE11, Legacy Edge, and older browsers
+                }
+            });
+        });
+
+    });
+   
+});
```
Comment (BY00565233) [inline src/elements/hx-popover/index.spec.js:74]: Please write a test for: 
 $onCreate () {
       this.DEFAULT_POSITION = 'bottom-right';
        this.POSITION_OFFSET = 20;
    }

File: src/elements/hx-popover/index.spec.js [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+                const shadow = component.shadowRoot.innerHTML;
+
+                expect(component).shadowDom.to.equal(shadow);
+            });
+
+            it('should render the Shadow Root mode open', async () => {
+                const component = /** @type { HXPopoverElement } */ await fixture(template);
+                const mode = component.shadowRoot.mode;
+
+                expect(mode).to.equal("open");
```
[the remark this answers is not in the document]
Comment (lalithkarikelli) [reply src/elements/hx-popover/index.spec.js]: Changes have been done, please review.

File: src/elements/hx-popover/index.spec.js (hunk 1 above)
Comment (lalithkarikelli) [reply src/elements/hx-popover/index.spec.js:85]: Changes have been done, please review.

Comment (100stacks) [reply src/elements/hx-popover/index.spec.js:85]: Please add a test for:

```js
this.POSITION_OFFSET = 20;
```

Comment (lalithkarikelli) [reply src/elements/hx-popover/index.spec.js:92]: Changes have been done, please review.

File: src/elements/hx-popover/index.spec.js [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+            expect(bottomStart).to.equal(defaultPosition);
+        });
+        it('test for position offset', async () => {
+            const component = /** @type {HXPopoverElement} */ await fixture(template);
+            const positionOffset = 20;
+            const offsetValue = component.POSITION_OFFSET;
+
+            expect(offsetValue).to.equal(positionOffset);
+        });
+    });
```
Comment (100stacks) [inline src/elements/hx-popover/index.spec.js:91]: Please add a test for the `setShadowPosition` method:

```js
    setShadowPosition (position) {
        this._elRoot.setAttribute('position', position);
    }
```

The test should test changing `position` to another supported value found here:

https://github.com/HelixDesignSystem/helix-ui/blob/master/src/utils/alignment/index.js#L30-L59

File: src/elements/hx-popover/index.spec.js [hunk 4]
```diff
@@ -0,0 +1,92 @@
+import { fixture, expect } from '@open-wc/testing';
```
Comment (100stacks) [inline src/elements/hx-popover/index.spec.js:1]: Please update this comment in `src/elements/hx-popover/index.js`:

```js
    /**
     * This position attribute is for the popover arrow located on the Shadow DOM element root.
     *
     * @override
     * @param {NormalizedPositionString}
     */
```

https://github.com/HelixDesignSystem/helix-ui/blob/master/src/elements/hx-popover/index.js#L39-L42

File: src/elements/hx-popover/index.spec.js (hunk 3 above)
Comment (lalithkarikelli) [reply src/elements/hx-popover/index.spec.js:91]: Changes have been done, please review.

File: src/elements/hx-popover/index.spec.js (hunk 4 above)
Comment (lalithkarikelli) [reply src/elements/hx-popover/index.spec.js:1]: Changes have been done, please review.

File: src/elements/hx-popover/index.spec.js [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+        });
+        it('test for position offset', async () => {
+            const component = /** @type {HXPopoverElement} */ await fixture(template);
+            const positionOffset = 20;
+            const offsetValue = component.POSITION_OFFSET;
+
+            expect(offsetValue).to.equal(positionOffset);
+        });
+        it('test for checking all positions', async () => {
+            const component = /** @type {HXPopoverElement} */ await fixture(template);
```
Comment (100stacks) [inline src/elements/hx-popover/index.spec.js:92]: We can test this method:

```js
setShadowPosition (position) {
        this._elRoot.setAttribute('position', position);
    }
```

And retrieve the results using something similiar to this:

```js
component.shadowRoot.querySelector('div').getAttribute('position')
```

Your solution will have sequential steps that follows our testing pattern.

Comment (lalithkarikelli) [reply src/elements/hx-popover/index.spec.js:92]: Changes have been done, please review.

**Pull request merged**